Source/WebCore/workers/service/server/SWServer.cpp

```diff
@@ -1346,22 +1336,35 @@ void SWServer::unregisterServiceWorkerClient(const ClientOrigin& clientOrigin, S
         iterator->value.terminateServiceWorkersTimer->startOneShot(m_isProcessTerminationDelayEnabled && !MemoryPressureHandler::singleton().isUnderMemoryPressure() && !shouldContextConnectionBeTerminatedWhenPossible && !didUnregister ? defaultTerminationDelay : 0_s);
     }
 
-    // If the app-bound value changed after this client was removed, we know it was the only app-bound
-    // client for its origin, and we should update all workers to reflect this.
-    auto appInitiatedValueAfter = clientIsAppInitiatedForRegistrableDomain(clientOrigin.clientRegistrableDomain());
-    if (appInitiatedValueBefore != appInitiatedValueAfter)
-        updateAppInitiatedValueForWorkers(clientOrigin, appInitiatedValueAfter);
+    if (shouldUpdateRegistrations == ShouldUpdateRegistrations::Yes) {
+        // If the app-bound value changed after this client was removed, we know it was the only app-bound
+        // client for its origin, and we should update all workers to reflect this.
+        auto appInitiatedValueAfter = clientIsAppInitiatedForRegistrableDomain(clientOrigin.clientRegistrableDomain());
+        if (appInitiatedValueBefore != appInitiatedValueAfter)
+            updateAppInitiatedValueForWorkers(clientOrigin, appInitiatedValueAfter);
+    }
 
     auto registrationIterator = m_clientToControllingRegistration.find(clientIdentifier);
     if (registrationIterator == m_clientToControllingRegistration.end())
         return;
 
-    if (RefPtr registration = m_registrations.get(registrationIterator->value))
-        registration->removeClientUsingRegistration(clientIdentifier); // May destroy the registration.
+    if (shouldUpdateRegistrations == ShouldUpdateRegistrations::Yes) {
```

Comment:
Is it feasible to only perform this check once in the function and gather the associated code inside one block?